### PR TITLE
fix: rename-contigs to include target bed intersection and add mapping files to resources

### PR DESCRIPTION
**Pull request by famosab**
- Avoiding errors with contigs not being present in the target file.
- Adding more elaborate rename files as resources.

### CODE REVIEW

**Comment by famosab** [comment]
I am intersection with the confidence regions and run into 
```
[E::faidx_adjust_position] The sequence "chrEBV" was not found
faidx_fetch_seq failed at chrEBV:68570
Error: line number 30405 of file /dev/fd/63 has 7 fields, but 11 were expected.
```

**Comment by famosab** [comment]
I tried. out `confidence` and `target` regions locally and I am not sure which one should be used. I changed the norm to pipe the result of the intersect into `bcftools norm` otherwise I always ran into the above error.

File: workflow/rules/common.smk [hunk 1]
```diff
@@ -229,11 +229,22 @@ def get_target_bed_statement(wildcards):
 
 
 def get_target_regions(wildcards):
```
Comment (famosab) [inline workflow/rules/common.smk:231]: Target and confidence regions probably do not need to be modified both.

File: workflow/rules/common.smk [hunk 2]
```diff
@@ -299,8 +310,13 @@ def get_stratified_truth(suffix=""):
 
 
 def get_confidence_regions(wildcards):
```
Comment (famosab) [inline workflow/rules/common.smk:312]: Target and confidence regions probably do not need to be modified both.

I would use the confidence regions.

File: workflow/rules/eval.smk [hunk 3]
```diff
@@ -60,15 +60,23 @@ rule remove_non_pass:
...
     output:
         "results/normalized-variants/{callset}.vcf.gz",
     log:
         "logs/normalize-calls/{callset}.log",
+    conda:
+        "../envs/tools.yaml"
+    shell:
+        "(bedtools intersect -b {input.regions} -a "
+        "<(bcftools view {input.bcf}) -wa -f 1.0 -header | "
+        "bcftools norm --atomize --check-ref s --fasta-ref {input.ref} --rm-dup exact | "
```
Comment (famosab) [inline workflow/rules/eval.smk:78]: Maybe we need to change back to using params and get_norm_params for this part?

Comment (BiancaStoecker) [reply workflow/rules/eval.smk]: Yes, get_norm_params contains an extra parameter for test cases where limit_reads is true

**Comment by famosab** [comment]
Also works with the Freiburg data (the VCFs containing `chrEBV` and `_decoy`).

File: workflow/rules/common.smk [hunk 4]
```diff
@@ -220,11 +220,22 @@ def get_target_bed_statement(wildcards):
...
+        if "target-regions" in benchmark:
+            return "resources/regions/{benchmark}/target-regions.bed"
+        else:
+            return []
     else:
-        return []
+        benchmark_name = config["variant-calls"][wildcards.callset]["benchmark"]
+        benchmark = get_benchmark(benchmark_name)
+        if "target-regions" in benchmark:
+            # TODO use f-string when this is fixed: https://github.com/snakemake/snakefmt/issues/215
```
Comment (johanneskoester) [inline workflow/rules/common.smk:233]: Isn't this fixed now? The issue is closed.

**Review by johanneskoester — CHANGES_REQUESTED** [review]

File: workflow/rules/common.smk (hunk 4 above)
Comment (famosab) [reply workflow/rules/common.smk]: I removed all TODOs and added `f`

**Pull request merged**